src/System.ComponentModel.TypeConverter/src/System/ComponentModel/AmbientValueAttribute.cs

```diff
@@ -139,26 +133,14 @@ public override bool Equals(object obj)
                 return true;
             }
 
-            AmbientValueAttribute other = obj as AmbientValueAttribute;
-
-            if (other != null)
+            if (!(obj is AmbientValueAttribute other))
             {
-                if (Value != null)
-                {
-                    return Value.Equals(other.Value);
-                }
-                else
-                {
-                    return (other.Value == null);
-                }
+                return false;
             }
-            return false;
-        }
 
-        public override int GetHashCode()
-        {
-            return base.GetHashCode();
+            return Value != null ? Value.Equals(other.Value) : other.Value == null;
```

Comment:
Why invert these?  I think it reads more clearly without the negation on the if condition and when other is used in the if block, e.g.
```C#
if (obj is AmbientValueAttribute other)
{
    return Value != null ? Value.Equals(other.Value) : other.Value == null;
}
return false;
```